devito/passes/iet/parpragma.py

```diff
@@ -566,7 +571,7 @@ def __init__(self, sregistry, options, platform, compiler):
         super().__init__(sregistry, options, platform, compiler)
 
         self.gpu_fit = options['gpu-fit']
-        self.par_tile = options['par-tile']
+        self.par_tile = UnboundTuple(options['par-tile'])
```

Comment:
Unsure , now, actually. Why isn't UnboundedMultiTuple(...) actually just usable here?

Reply:
`UnboundedMultiTuple` does not itearte infinitely over each tuple and throws an error after reaching the last element of a subtuple instead of returning it forever, I assume this is intended behaviour so did not want to modify it

Reply:
you could easily add

```
self.par_tile.next(default='last')
```

or even

```
umt = UnboundMultiTuple(..., default='last')
```

to change default behavior

I **think** what we want here is an UnboundMultiTuple to actually support the user providing `par-tile: ((16, 16, 16), (4, 4, 4), (16, 4, 7), ...)`

Reply:
I still think that two classes for unbounded tuples is overkill, but I also appreciate that at this stage you want to move on...